### PR TITLE
add additional scrape jobs for prometheus

### PR DESCRIPTION
**Pull request by guerremdq**
## Description

We want to scrape other services like ie: Push gateway running on the same cluster and we need to be able to modify the Prometheus configuration. 

is this something that could be supported by Astronomer?

### CODE REVIEW

**Comment by danielhoherd** [comment]
@guerremdq thanks for the submission! Please add `additionalScrapeJobs: {}` to `charts/prometheus/values.yaml` so the new feature is more discoverable. Also we need to test the new feature, so please add a new `test_additional_scrape_jobs` test case for it in `tests/chart_tests/test_prometheus_config_configmap.py`. There are a bunch of test cases in there that would be great as a copy/paste way to start the new test case. If you need help getting started, here is a short tutorial https://github.com/astronomer/astronomer/blob/master/tests/chart_tests/README.md and I'm free to answer any questions, review code, and give suggestions.

**Comment by guerremdq** [comment]
Hey @danielhoherd , Thanks for the quick reply on this topic. I have added the test however I was unable to run them. Are these tests running on a CI?

File: tests/chart_tests/test_prometheus_config_configmap.py [hunk 1]
```diff
@@ -305,3 +305,41 @@ def assert_relabel_config_for_non_auto_generated_namesaces(self, chart):
...
+        found_static_job = False
+        found_kubernetes_job = False
+        for job in prometheus_config["scrape_configs"]:
+            if job["job_name"] == "example-static-job":
+                found_static_job = True
+            elif job["job_name"] == "example-kubernetes-job":
+                found_kubernetes_job = True
+
+        assert found_static_job, "Static job not found in rendered ConfigMap"
+        assert found_kubernetes_job, "Kubernetes job not found in rendered ConfigMap"
```
Comment (danielhoherd) [inline tests/chart_tests/test_prometheus_config_configmap.py:345]: Awesome job on the tests! I found a way to simplify the test you wrote while doing a more exact comparison between the inputs and outputs:
```suggestion
    def test_additional_scrape_jobs(self, kube_version):
        static_job = {
            "job_name": "example-static-job",
            "static_configs": [{"targets": ["localhost:9090"]}],
        }
        kubernetes_job = {
            "job_name": "example-kubernetes-job",
            "kubernetes_sd_configs": [
                {
                    "role": "endpoints",
                    "namespaces": {"names": ["default"]},
                }
            ],
        }
        doc = render_chart(
            kube_version=kube_version,
            show_only=self.show_only,
            name="astronomer",
            values={
                "prometheus": {
                    "additionalScrapeJobs": [
                        static_job,
                        kubernetes_job,
                    ]
                }
            },
        )[0]
        scrape_configs = yaml.safe_load(doc["data"]["config"])["scrape_configs"]

        assert static_job in scrape_configs
        assert kubernetes_job in scrape_configs
```

**Comment by guerremdq** [comment]
@danielhoherd We are currently running `0.32.4` can this maybe cherry-picked and released under 0.32 too?

**Comment by danielhoherd** [comment]
@guerremdq yeah, this is something that is easy to include in all supported versions, which would be 0.32, 0.33, 0.34. https://docs.astronomer.io/software/release-lifecycle-policy#stable

**Comment by danielhoherd** [comment]
Merged to master, merged master to release-0.34, cherry-picked change to release-0.33 and release-0.32

**Comment by guerremdq** [comment]
Hey @danielhoherd when we can expect the new release ?

**Pull request merged**